ingestion/src/metadata/cli/ingest_dbt.py

```diff
@@ -0,0 +1,210 @@
+#  Licensed under the Collate Community License, Version 1.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/LICENSE
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""
+DBT Artifacts Ingestion CLI module
+"""
+
+import sys
+import traceback
+from pathlib import Path
+from typing import Dict
+
+import yaml
+from metadata.utils.logger import cli_logger
+from metadata.workflow.metadata import MetadataWorkflow
+
+logger = cli_logger()
+
+
+def find_dbt_project_config(dbt_project_path: Path) -> Dict:
+    """
+    Find and load dbt_project.yml configuration
+    
+    :param dbt_project_path: Path to the dbt project directory
+    :return: Parsed dbt project configuration
+    """
+    dbt_project_file = dbt_project_path / "dbt_project.yml"
+    
+    if not dbt_project_file.exists():
+        raise FileNotFoundError(f"dbt_project.yml not found in {dbt_project_path}")
+        
+    try:
+        with open(dbt_project_file, 'r') as file:
+            return yaml.safe_load(file)
+    except Exception as exc:
+        raise ValueError(f"Failed to parse dbt_project.yml: {exc}")
+
+
+def extract_openmetadata_config(dbt_config: Dict) -> Dict:
+    """
+    Extract OpenMetadata connection configuration from dbt project config
+    
+    :param dbt_config: Parsed dbt project configuration
+    :return: OpenMetadata configuration
+    """
+    vars_config = dbt_config.get('vars', {})
+    
+    # Look for OpenMetadata configuration in vars (only standard naming)
+    host_port = vars_config.get('openmetadata_host_port')
+    jwt_token = vars_config.get('openmetadata_jwt_token')
+    service_name = vars_config.get('openmetadata_service_name')
+    
+    if not jwt_token or not host_port or not service_name:
+        missing_vars = []
+        if not host_port:
+            missing_vars.append('openmetadata_host_port')
+        if not jwt_token:
+            missing_vars.append('openmetadata_jwt_token')
+        if not service_name:
+            missing_vars.append('openmetadata_service_name')
+            
+        raise ValueError(
+            f"Required OpenMetadata configuration not found in dbt_project.yml vars.\n"
+            f"Missing variables: {', '.join(missing_vars)}\n"
+            f"Please add the following to your dbt_project.yml:\n"
+            f"vars:\n"
+            f"  openmetadata_jwt_token: 'your-jwt-token'\n"
+            f"  openmetadata_host_port: 'your-host-port (e.g. http://openmetadata-server:8585/api)'\n"
+            f"  openmetadata_service_name: 'your-service-name'"
+        )
```

Comment:
instead of doing this manual validation lets create a pedantic model and just parse the dictionary into the object